src/trio/_core/_run.py

```diff
@@ -103,8 +103,7 @@ class _NoStatus(metaclass=NoPublicConstructor):
 
 # Decorator to mark methods public. This does nothing by itself, but
 # trio/_tools/gen_exports.py looks for it.
-# Explicit "Any" is not allowed
-def _public(fn: FnT) -> FnT:  # type: ignore[misc]
+def _public(fn: Callable[PS, RetT]) -> Callable[PS, RetT]:
```

Comment:
Actually, this should stay as a typevar `bound=Callable[..., object]`. There's a subtle difference between that and paramspec - the bound typevar means the callable must be exactly the same from input to output, so things like overloads are preserved. This still might work if type checkers special case it, but it's not clear in the spec how it'll interact with ParamSpec.

Reply:
I was considering changing to `def _public(fn: T) -> T:` because that's all it really does, and I think this cements that that might be a better solution.

Reply:
Changed in 55964ad